string[] pose_names
int32[] pose_types